File: miraculous-frontend/miraculous-ui/src/dto/full-album-info-dto/full-album-info-dto.ts
export class FullAlbumInfoDto {

    id: number;
    title: string;
    artistId: number;
    artistLogin: string;
    dateOfRelease: string;
    numberOfTracks: number;
    albumDuration: number;
    genres: string[];
    description: string;
    isFree: boolean;
    price: number;
    numberOfLikes: number;
    coverPictureUrlName: string;
    size: number;
    inCartOfCurrentVisitor: boolean;
    likedByCurrentVisitor: boolean;


  constructor(id?: number, title?: string, artistId?: number, artistLogin?: string, dateOfRelease?: string,
              numberOfTracks?: number, albumDuration?: number, genres?: string[], description?: string,
              isFree?: boolean, price?: number,
              numberOfLikes?: number,
              coverPictureUrl?: string,
              size?: number,
              isInCurrentVisitorCart?: boolean,
              isLikedByCurrentVisitor?: boolean) {

    this.id = id;
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.dateOfRelease = dateOfRelease;
    this.numberOfTracks = numberOfTracks;
    this.albumDuration = albumDuration;
    this.genres = genres;
    this.description = description;
    this.isFree = isFree;
    this.price = price;
    this.numberOfLikes = numberOfLikes;
    this.coverPictureUrlName = coverPictureUrl;
    this.size = size;
    this.inCartOfCurrentVisitor = isInCurrentVisitorCart;
    this.likedByCurrentVisitor = isLikedByCurrentVisitor;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/FullAlbumInfoDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FullAlbumInfoDto {

    private Long id;
    private String title;
    private Long artistId;
    private String artistLogin;
    private String dateOfRelease;
    private Integer numberOfTracks;
    private Double albumDuration;
    private List<String> genres;
    private String description;
    private Boolean isFree;
    private Double price;
    private Long numberOfLikes;
    private String coverPictureUrlName;
    private Double size;
    private boolean isLikedByCurrentVisitor;
    private boolean isInCartOfCurrentVisitor;

}
